java-checks/src/main/java/org/sonar/java/checks/MembersDifferOnlyByCapitalizationCheck.java

```diff
@@ -203,4 +228,28 @@ private static boolean isVariableToExtract(Symbol symbol) {
   private static boolean isMethodToExtract(Symbol symbol) {
     return symbol.isMethodSymbol() && !"<init>".equals(symbol.name());
   }
+
+  private static class ReturnVisitor extends BaseTreeVisitor {
+
+    private final Symbol variableSymbol;
+    private boolean returnsVariable;
+    private int returnCount;
+
+    ReturnVisitor(Symbol variableSymbol) {
+      this.variableSymbol = variableSymbol;
+    }
+
+    @Override
+    public void visitReturnStatement(ReturnStatementTree tree) {
+      returnCount++;
+      ExpressionTree returnExpression = tree.expression();
+      if (returnExpression != null && returnExpression.is(Tree.Kind.IDENTIFIER)) {
+        returnsVariable = ((IdentifierTree) returnExpression).symbol().equals(variableSymbol);
+      }
+    }
+
+    boolean singleReturnWithVariableSymbol() {
+      return returnCount == 1 && returnsVariable;
```

Comment:
OK, so it implicitly means that you can have `n` other statements in the body of the method, as long as you return the variable, it's ok? Is it what you really want to achieve? (can't see a test case related to that).

If you want to target explicitly single-return-statement methods, it should be easier to simply look at the body, which has to be of size 1, with a single return statement returning the variable. And no need for a visitor for that.